core/src/sumeragi/fault.rs

```diff
@@ -38,866 +37,1337 @@ impl FaultInjection for NoFault {
-    type GenesisNetwork = G;
-
-    fn from_configuration(
-        configuration: &Configuration,
-        events_sender: EventsSender,
-        wsv: Arc<WorldStateView>,
-        transaction_validator: TransactionValidator,
-        telemetry_started: bool,
-        genesis_network: Option<G>,
-        queue: Arc<Queue>,
-        broker: Broker,
-        kura: Arc<Kura>,
-        network: Addr<IrohaNetwork>,
-    ) -> Result<Self> {
-        let network_topology = Topology::builder()
-            .at_block(EmptyChainHash::default().into())
-            .with_peers(configuration.trusted_peers.peers.clone())
-            .build()?;
-
-        Ok(Self {
-            key_pair: configuration.key_pair.clone(),
-            topology: network_topology,
-            peer_id: configuration.peer_id.clone(),
-            voting_block: None,
-            votes_for_blocks: BTreeMap::new(),
-            events_sender,
-            wsv,
-            txs_awaiting_receipts: HashMap::new(),
-            txs_awaiting_created_block: HashSet::new(),
-            votes_for_view_change: HashMap::new(),
-            commit_time: Duration::from_millis(configuration.commit_time_limit_ms),
-            tx_receipt_time: Duration::from_millis(configuration.tx_receipt_time_limit_ms),
-            block_time: Duration::from_millis(configuration.block_time_ms),
-            block_height: 0,
-            invalidated_blocks_hashes: Vec::new(),
-            telemetry_started,
-            transaction_limits: configuration.transaction_limits,
-            transaction_validator,
-            genesis_network,
-            queue,
-            broker,
-            kura,
-            network,
-            actor_channel_capacity: configuration.actor_channel_capacity,
-            fault_injection: PhantomData,
-            gossip_batch_size: configuration.gossip_batch_size,
-            gossip_period: Duration::from_millis(configuration.gossip_period_ms),
-        })
+impl<F: FaultInjection> Debug for SumeragiWithFault<F> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Sumeragi")
+            .field("public_key", &self.key_pair.public_key())
+            .field("peer_id", &self.peer_id)
+            .finish()
     }
 }
 
-#[async_trait::async_trait]
-impl<G: GenesisNetworkTrait, F: FaultInjection> Actor for SumeragiWithFault<G, F> {
-    fn actor_channel_capacity(&self) -> u32 {
-        self.actor_channel_capacity
-    }
-
-    async fn on_start(&mut self, ctx: &mut Context<Self>) {
-        self.broker.subscribe::<Init, _>(ctx);
-        self.broker.subscribe::<Message, _>(ctx);
-        self.broker.subscribe::<CommitBlock, _>(ctx);
-        self.broker.subscribe::<NetworkMessage, _>(ctx);
-    }
+// TODO: In general naming things after the programming patterns is
+// considered a bad practice. We need a better name.
+/// Internal structure that retains the state.
+pub struct SumeragiStateMachineData {
```

Comment:
`SumeragiState`?

Reply:
`State` is enough. It's no more a state machine than other things we don't have a state machine inside of. Moreover, now the fully qualified path is `sumeragi::main_loop::State`